### PR TITLE
completed task 11 get articles with a topic query

### PR DESCRIPTION
**Pull request by Arkodia0001**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -414,3 +405,25 @@ describe("GET /api/users", () => {
...
+
+describe("GET /api/articles (topic query)", () => {
+    test("should return an array of articles that match the given topic query", () => {
+        return request(app).get('/api/articles?topic=cats').expect(200).then(({body: {articles}}) => {
+            expect(articles.length).toBe(1)
+            articles.forEach((article) => {
+                expect(article.topic).toBe('cats')
+            })
+        })
+    })
```
Comment (MRombi) [inline __tests__/app.test.js:417]: We should add a test to ensure that when our topic is paper that does not have any articles, the endpoint sends an empty array

File: app/models/nc-news.model.js [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
-    FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+exports.selectArticles = (topic) => {
+  const validTopics = ["cats", "mitch", "paper"];
```
Comment (MRombi) [inline app/models/nc-news.model.js:13]: This works here but if we were to add another topic you'd need to modify this, it would not be dynamic and might break the app

**Review by MRombi — COMMENTED** [review]
Please take into account my feedback before moving forward
